Admin/BaseMediaAdmin.php

```diff
@@ -106,7 +106,7 @@ public function getNewInstance()
 
         if ($this->hasRequest()) {
             if ($this->getRequest()->isMethod('POST')) {
-                $media->setProviderName($this->getRequest()->get(sprintf('%s[providerName]', $this->getUniqid()), null, true));
+                $media->setProviderName($this->getRequest()->get(sprintf('%s', $this->getUniqid()), null, true)['providerName']);
```

Comment:
Argh, you're right about http-foundation vs http-kernel! Gotta find another way to detect the version…

> I believe the parameter of deep = true in this case already triggers an error if the app is using http-foundation <=2.8. 

I don't understand why. The parameter is present in <= 2.8, why would it trigger an error.

> Then Why don't we just remove them all and the problem will be solved without extra checks ?

If you find a way that works with both 2.7, 2.8, 3.0 without any deprecation message being thrown be my guest.